src/ir/context.rs

```diff
@@ -569,6 +569,24 @@ impl<'ctx> BindgenContext<'ctx> {
         self.used_template_parameters = Some(used_params);
     }
 
+    /// Return `true` if `item` uses the given `template_param`, `false`
+    /// otherwise.
+    ///
+    /// This method may only be called during the codegen phase, because the
+    /// template usage information is only computed as we enter the codegen
+    /// phase.
+    pub fn uses_template_parameter(&self, item: ItemId, template_param: ItemId) -> bool {
+        assert!(self.in_codegen_phase(),
```

Comment:
Yeah, but also something we need pretty badly IMO. The transmuted lifetime can make it easy to accidentally have conflicting lifetimes for no good reason. I spent probably a half hour to an hour trying to get the named template parameter analysis working with the made up lifetime :-/